gsheets4/gsheets_utils.bal

```
@@ -15,8 +15,44 @@
 // under the License.
 
 import ballerina/http;
+import ballerina/io;
 
 function setResponseError(json jsonResponse) returns error {
-    error err = error(SPREADSHEET_ERROR_CODE, { message: jsonResponse.message.toString() });
+    return error(SPREADSHEET_ERROR_CODE, { message: jsonResponse["error"].message.toString()});
+}
+
+function setResError(error apiResponse) returns error {
+    return error(SPREADSHEET_ERROR_CODE, { message: <string>apiResponse.detail().message});
+}
+
+function setJsonResponse(json jsonResponse, int statusCode) returns Spreadsheet|error {
+    if (statusCode == http:OK_200) {
+        Spreadsheet spreadsheet = convertToSpreadsheet(jsonResponse);
+        return spreadsheet;
+    }
+    return setResponseError(jsonResponse);
+}
+
+function setResponse(json jsonResponse, int statusCode) returns boolean|error{
+    if (statusCode == http:OK_200) {
+        return true;
+    }
+    return setResponseError(jsonResponse);
+}
+
+function validateSpreadSheetId(string spreadSheetId) returns error? {
+    string regEx = "[a-zA-Z0-9-_]+";
+    boolean isMatch = checkpanic spreadSheetId.matches(regEx);
```

Here again do we need abrupt termination? Shouldn't we just return the error?